examples/src/fungible-token.js

```diff
@@ -1,87 +1,136 @@
-import {
-  NearBindgen,
-  call,
-  view,
-  initialize,
-  near,
-  LookupMap,
-  assert,
-} from "near-sdk-js";
+import { NearBindgen, call, view, initialize, near, LookupMap, assert } from "near-sdk-js";
 
 @NearBindgen({ initRequired: true })
 export class FungibleToken {
-  constructor() {
-    this.accounts = new LookupMap("a");
-    this.totalSupply = 0n;
-  }
+  accounts = new LookupMap("a");
+  accountRegistrants = new LookupMap("r");
+  accountDeposits = new LookupMap("c");
+  totalSupply = "0";
 
   @initialize({})
-  init({ prefix, totalSupply }) {
-    this.accounts = new LookupMap(prefix);
-    this.totalSupply = BigInt(totalSupply);
-    this.accounts.set(near.signerAccountId(), this.totalSupply);
-    // In a real world Fungible Token contract, storage management is required to denfense drain-storage attack
+  init({ owner_id, total_supply }) {
+    assert(BigInt(total_supply) > BigInt(0), "Total supply should be a positive number");
+    assert(this.totalSupply === "0", "Contract is already initialized");
+    this.totalSupply = total_supply;
+    this.accounts.set(owner_id, this.totalSupply);
+  }
+
+  internalGetMaxAccountStorageUsage() {
```

Comment:
Do we really need to actually add and delete acc to get max storage?
Are we mimicking rust logic here, is it done there the same way?
If we really need to get storage usage, why not pass accId and require the user to attach less N for shorter accs?

Reply:
> Do we really need to actually add and delete acc to get max storage?

There is a way by getting the cost per character and then multiplying by max length of accountId (64 chars), but this implementation is also seen here: 

https://github.com/near/near-sdk-rs/blob/1a39f1e2406517ee93bc4efb0b859751297e645d/near-contract-standards/src/fungible_token/core_impl.rs#L50-L56

Reply:
Implemented dynamic storage on accId length

Reply:
Cool!
I will clarify, why storage logic is so complex.